internal/provider/table_schema_resource.go

```diff
@@ -26,12 +26,19 @@ func NewTableSchemaResource() resource.Resource {
 
 }
 
-type fieldSpec struct {
+type metricFieldSpec struct {
 	Name     string              `tfsdk:"name"`
 	DataType string              `tfsdk:"data_type"`
 	NotNull  basetypes.BoolValue `tfsdk:"not_null"`
 }
 
+type dimensionFieldSpec struct {
+	Name             string              `tfsdk:"name"`
+	DataType         string              `tfsdk:"data_type"`
+	NotNull          basetypes.BoolValue `tfsdk:"not_null"`
+	SingleValueField basetypes.BoolValue `tfsdk:"single_value_ield"`
```

Comment:
Small spelling mistake here, should be: "single_value_field"

Reply:
We also need, now that we've added the single_value_field property, to add that to the terraform resource schema here:

```go
"single_value_field": schema.BoolAttribute{
  Description: "Whether the dimension is a single value field.",
  Optional:    true,
},
```